### ISSUE TITLE
Parallel pktgen build fails because of absent lua and common libraries

### ISSUE THREAD

**Issue by gshimansky** [pktgen/Pktgen-DPDK#102]
I run "make -j 88" since I have 88 logical cores on my server and nproc command returns this value. With this option my pktgen build fails like this:

> LD pktgen
> /usr/bin/ld: cannot find -llua
> /usr/bin/ld: cannot find -lcommon
> collect2: error: ld returned 1 exit status
> /localdisk/gashiman/go/src/github.com/intel-go/yanff/dpdk/dpdk-stable-17.02.1/mk/rte.app.mk:246: 
> recipe for target 'pktgen' failed
> make[4]: *** [pktgen] Error 1
> /localdisk/gashiman/go/src/github.com/intel-go/yanff/dpdk/dpdk-stable-17.02.1/mk/rte.extapp.mk:42: recipe for target 'all' failed
> make[3]: *** [all] Error 2
> /localdisk/gashiman/go/src/github.com/intel-go/yanff/dpdk/dpdk-stable-17.02.1/mk/rte.extsubdir.mk:46: recipe for target 'app' failed
> make[2]: *** [app] Error 2
> make[2]: *** Waiting for unfinished jobs....

I can see lua library building after this message, so I think it is necessary to add necessary dependencies. Make: GNU Make 4.1 on Fedora 25. I am building pktgen 3.1.0.

**Comment by pktgen** [comment]
> On Jun 15, 2017, at 11:21 AM, Gregory Shimansky <notifications@github.com> wrote:
> 
> I run "make -j 88" since I have 88 logical cores on my server and nproc command returns this value. With this option my pktgen build fails like this:

I have not setup Pktgen build in parallel, but I do take patches for it :-)

> 
> LD pktgen
> /usr/bin/ld: cannot find -llua
> /usr/bin/ld: cannot find -lcommon
> collect2: error: ld returned 1 exit status
> /localdisk/gashiman/go/src/github.com/intel-go/yanff/dpdk/dpdk-stable-17.02.1/mk/rte.app.mk:246:
> recipe for target 'pktgen' failed
> make[4]: *** [pktgen] Error 1
> /localdisk/gashiman/go/src/github.com/intel-go/yanff/dpdk/dpdk-stable-17.02.1/mk/rte.extapp.mk:42: recipe for target 'all' failed
> make[3]: *** [all] Error 2
> /localdisk/gashiman/go/src/github.com/intel-go/yanff/dpdk/dpdk-stable-17.02.1/mk/rte.extsubdir.mk:46: recipe for target 'app' failed
> make[2]: *** [app] Error 2
> make[2]: *** Waiting for unfinished jobs....
> 
> I can see lua library building after this message, so I think it is necessary to add necessary dependencies. Make: GNU Make 4.1 on Fedora 25.
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
> 

Keith Wiles
keith.wiles@icloud.com

**Comment by gshimansky** [comment]
I think you forgot to mention libraries as an app dependency. Looking at the complex build system from DPDK I think the correct way to do it is to add a line like this

```
diff -Naur pktgen-v3.1.0/app/Makefile pktgen-v3.1.0-fixed/app/Makefile
--- pktgen-v3.1.0/app/Makefile  2016-12-09 16:43:44.000000000 -0600
+++ pktgen-v3.1.0-fixed/app/Makefile    2017-06-15 15:07:37.528358051 -0500
@@ -66,6 +66,7 @@

 MYLIB_PATH := -L$(LUA_LIB) -L$(COMMON_LIB)
 MYLIBS := -llua -lcommon
+LDLIBS_PATH := $(LUA_LIB) $(COMMON_LIB)

 ifeq ($(CONFIG_RTE_BUILD_SHARED_LIB),y)
        MYLIBS += -lrte_pmd_ixgbe
```
With this change I can successfully build pktgen 3.1.0 with -j 88.

**Comment by pktgen** [comment]
Thanks I will have a look
